### PR TITLE
[AVRO-2579]: Fix test by sorting fields for deterministic order

### PR DESCRIPTION
**Pull request by contextshuffling**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Avro Jira](https://issues.apache.org/jira/browse/AVRO/) issues 
   - https://issues.apache.org/jira/projects/AVRO/issues/AVRO-2579
### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
   - modifies the unit tests to make it more stable

### Commits

- [x] My commits all reference Jira issues in their subject lines

### CODE REVIEW

File: lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java [hunk 1]
```diff
@@ -728,7 +729,14 @@ public static Schema makeNullable(Schema schema) {
     do {
       if (excludeJava && c.getPackage() != null && c.getPackage().getName().startsWith("java."))
         break; // skip java built-in classes
-      for (Field field : c.getDeclaredFields())
+      Field[] declaredFields = c.getDeclaredFields();
+      Arrays.sort(declaredFields, new Comparator<Field>() {
```
Comment (Fokko) [inline lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java]: We can also use the shorthand:
```suggestion
      Arrays.sort(declaredFields, Comparator.comparing(Field::getName));
```

**Comment by Pezhin** [comment]
CI fails for weird issues. The code passes locally.

**Comment by contextshuffling** [comment]
CI failed for weird reason. The test passes locally

**Comment by Fokko** [comment]
Thanks @contextshuffling 

I've restarted the CI, sometimes Travis has some intermittent issues.

**Comment by contextshuffling** [comment]
@Fokko any follow-ups on this PR?

**Comment by Fokko** [comment]
Looks like that Travis had some network issues, I've restarted the CI (again).

**Comment by contextshuffling** [comment]
Potentially related: https://github.com/apache/pulsar/pull/6247

**Comment by Fokko** [comment]
Thanks @contextshuffling. Can you rebase against latest master?

**Comment by contextshuffling** [comment]
@Fokko Sorry for the late reply. should be good now

**Comment by Fokko** [comment]
We got a failing master again. Normally this doesn't happen that often, but I guess you are very unlucky @contextshuffling I'll check if I can fix master, let me get back to you.

**Comment by contextshuffling** [comment]
@Fokko I saw in the log that `ImportError: cannot import name 'ContextManager'`. First of all, this is in python which I do not see how my changes would touch it. After some search, this looks to be something newly added after py3.6. Is the CI image out-of-dated? Above it, the project itself builds and run all the test successfully and then the CI crashed due to this python script.

**Comment by Fokko** [comment]
@contextshuffling sorry for the late reply. Can you rebase onto master? Current master is green again.

**Pull request merged**